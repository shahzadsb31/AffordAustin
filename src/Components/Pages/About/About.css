.App {
  text-align: center;
}

.section_header {
  color: black;
  font-size: max(1em, 40px);
  font-weight: bolder;
  padding-top: 1em;
  margin-bottom: .5em;
}

.temp_interesting {
  color: black;
  font-size: max(.9em, 30px);
  margin-bottom: .5em;
}

.tools_apis {
  width: 85%;
  font-size: 28px;
}

.instance_data {
  width: 85%;
  font-size: 28px;
}

.about_text {
  font-size: max(.7em, 24px);
  width: 95%;
}



.about_list {
  margin-bottom: 10px;
}

.about_list:hover {
  box-shadow: rgba(0, 0, 0, 0.1) 0px 5px 15px;
}

.tool_about_list {
  margin-bottom: 10px;
  min-width: 280px;
  max-width: 300px;
  height: 440px;
  background-color: white;
  
}

.tool_desc {
  color: inherit !important;
  text-decoration: none !important;

}

.tool_about_list:hover {
  /* transform: scale(1.03); */
  box-shadow: rgba(0, 0, 0, 0.1) 0px 5px 15px;
}

.tool_logo {
  height: 150px;
  width: auto;
}