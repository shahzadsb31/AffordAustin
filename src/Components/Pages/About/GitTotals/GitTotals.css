.git_totals {
  min-width: max(6em, 200px);
  background-color: white;
  color: black;
  border-radius: 3rem;
  max-width: 9em;
}
