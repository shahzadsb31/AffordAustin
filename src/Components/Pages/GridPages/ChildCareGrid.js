import './Grid.css'
import 'bootstrap/dist/css/bootstrap.min.css'
import { Container, Row, Col, Spinner } from 'react-bootstrap';
import React, { useState, useEffect, useCallback } from 'react';
import FSBar from './FSBar/FSBar';
import axios from 'axios';
import ChildCareInstanceCard from './InstanceCards/ChildCareInstanceCard';
import { useSearchParams } from "react-router-dom";

const ChildCareGrid = () => {
    const [searchParams, setSearchParams] = useSearchParams();
    const [programs, setPrograms] = useState([]);
    const [loading, setLoading] = useState(true);
    const [totalNumPrograms, setTotalNumPrograms] = useState(1);
    const [programsPerPage, setProgramsPerPage] = useState(21);
    const [query, setQuery] = useState(searchParams.get('full_query') ?? "");  
    const [searchKeys, setSearchKeys] = useState([]);
    const [currentPage, setCurrentPage] = useState(searchParams.get('page') ?? 1);

    const getChildCareData = useCallback (async () => {
        setLoading(true);
        let endpoint = `https://api.affordaustin.me/api/childcare?page[size]=${programsPerPage}&page[number]=${currentPage}`;
        endpoint += (query === "") ? "" : "&" + query;
        const data = await axios.get(endpoint);
        setTotalNumPrograms(data.data.metadata.total_count);
        setPrograms(data.data.attributes);
        setLoading(false);
    }, [currentPage, programsPerPage, query]);

    useEffect(() => {
        getChildCareData();
    }, [searchParams, currentPage, query, searchKeys, getChildCareData]);

    const paginate = (pageNum) => {
        setSearchParams({page: pageNum});
        setCurrentPage(pageNum);
    };

    const getQuery = (new_query, new_search_query) => {
        let full_query = new_query;
        if (full_query !== "" && new_search_query !== ""){
            full_query += "&";
        }
        full_query += new_search_query;
        setQuery(full_query);
        let search_query = (new_search_query === "") ? [] : new_search_query.slice(7).split(" ");
        setSearchKeys(search_query);
        setSearchParams({page: 1});
        if (full_query !== query) {
            setCurrentPage(1);
        }

    };

    return (
        <div style={{ backgroundColor: "#f0f2f5" }}>
            <div className='grid mx-auto'>
                <Container fluid>
                    <Row>
                        <h1 className='grid_header'>Child Care</h1>
                    </Row>
                    <Row>
                    <FSBar totalInstances={totalNumPrograms} pageLimit={programsPerPage} paginate={paginate} currentPage={currentPage} sendQuery={getQuery} initialQuery={query} model="Childcare"/>
                    </Row>
                    <Row className="justify-content-center">
                        {loading ? <Spinner animation='border' role="status"/> : <></>}
                    </Row>
                    <Row className="g-3 justify-content-center" xs='auto'>
                        {loading ? <></> : programs.map(program => {
                            return (
                            <Col key={program.id}>
                                <ChildCareInstanceCard child_care={program} id={program.id} search_keys={searchKeys}/>
                            </Col>);
                        })}
                    </Row>
                </Container>
            </div>
        </div>
    );
};

export default ChildCareGrid;