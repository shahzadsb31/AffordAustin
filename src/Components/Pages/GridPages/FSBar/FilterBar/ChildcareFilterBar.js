// import './FilterBar.css'
import 'bootstrap/dist/css/bootstrap.min.css'
import React, { useState, useEffect } from 'react';
import { Row, Col, Form } from 'react-bootstrap';

//Change values to queries
const ChildcareFilterBar = ({sendQuery, initialQuery}) => {
    const x = {
            'ZipcodeFilter': '', 
            'CountyFilter': '',
            'HoursFilterStarting': '',
            'HoursFilterEnding': '', 
            'AgesFilter': ''
        };

    const [form, setForm] = useState(x);
    
    const setField = (field, value) => {
        setForm({
            ...form,
            [field]: value
        })
    }

    useEffect(() => {
        let isNotEmpty = form['ZipcodeFilter'] !== '';
        let filterQuery = form['ZipcodeFilter'];

        if ( isNotEmpty && form['CountyFilter'] !== '') {
            filterQuery += '&';
        }
        filterQuery += form['CountyFilter'];
        isNotEmpty = isNotEmpty || form['CountyFilter'] !== '';

        if (isNotEmpty && form['HoursFilterStarting'] !== '') {
            filterQuery += '&';
        }
        filterQuery += form['HoursFilterStarting'];
        isNotEmpty = isNotEmpty || form['HoursFilterStartin'] !== '';

        if (isNotEmpty && form['HoursFilterEnding'] !== '') {
            filterQuery += '&';
        }
        filterQuery += form['HoursFilterEnding'];
        isNotEmpty = isNotEmpty || form['HoursFilterEnding'] !== '';

        if ( isNotEmpty && form['AgesFilter'] !== '') {
            filterQuery += '&';
        }
        filterQuery += form['AgesFilter'];
        
        sendQuery(filterQuery);
    }, [form])

    return (
        <div style={{ textAlign:'center' }}>
            <h3>Filters:</h3>
            <Form>
                <Row className="g-3 justify-content-center" xs='auto'>
                    <Form.Group controlId='ZipcodeFilter' as={Col}>
                        <Form.Label>Zip Code</Form.Label>
                        <Form.Control
                            type='text'
                            placeholder='Find center near'
                            className='filter_text'
                            // From https://stackoverflow.com/questions/34223558/enter-key-event-handler-on-react-bootstrap-input-component
                            onKeyPress={e => {
                                if (e.key === "Enter") {
                                    setField('ZipcodeFilter', (e.target.value == "") ? "" : "zip_code=" + e.target.value);
                                }
                            }}
                        >
                        </Form.Control>
                    </Form.Group>
                    <Form.Group controlId='CountyFilter' as={Col}>
                        <Form.Label>County</Form.Label>
                        <Form.Control
                            as="select"
                            className='filter_select'
                            onChange={e => setField('CountyFilter', e.target.value)}
                            defaultValue={form['CountyFilter']}
                        >
                            <option value=''>Select County</option>
                            <option value='county=Bastrop'>Bastrop</option>
                            <option value='county=Caldwell'>Caldwell</option>
                            <option value='county=Hays'>Hays</option>
                            <option value='county=Travis'>Travis</option>
                            <option value='county=Williamson'>Williamson</option>
                        </Form.Control>
                    </Form.Group>
                    <Form.Group controlId='HoursFilter' as={Col}>
                        <Form.Label>Hours of Operation</Form.Label>
                        <Row>
                            <Col>
                                <Form.Select
                                    className='filter_select'
                                    onChange={e => setField('HoursFilterStarting', e.target.value)}
                                >
                                    <option value=''></option>
                                    <option value='start_hours_val=6'>6 am</option>
                                    <option value='start_hours_val=7'>7 am</option>
                                    <option value='start_hours_val=8'>8 am</option>
                                    <option value='start_hours_val=9'>9 am</option>
                                    <option value='start_hours_val=14'>2 pm</option>
                                </Form.Select>
                            </Col>
                            <Col>
                                <Form.Select
                                    className='filter_select'
                                    onChange={e => setField('HoursFilterEnding', e.target.value)}
                                >
                                    <option value=''></option>
                                    <option value='end_hours_val=13'>1 pm</option>
                                    <option value='end_hours_val=14'>2 pm</option>
                                    <option value='end_hours_val=15'>3 pm</option>
                                    <option value='end_hours_val=16'>4 pm</option>
                                    <option value='end_hours_val=17'>5 pm</option>
                                    <option value='end_hours_val=18'>6 pm</option>
                                </Form.Select>
                            </Col>
                        </Row>
                    </Form.Group>
                    <Form.Group controlId='AgesFilter' as={Col}>
                        <Form.Label>Age Group</Form.Label>
                        <Form.Select 
                            className='filter_select'
                            onChange={e => setField('AgesFilter', e.target.value)}
                        >
                            <option value=''>Select Age Group</option>
                            <option value='licensed_to_serve_ages=Infant'>Infant</option>
                            <option value='licensed_to_serve_ages=Toddler'>Toddler</option>
                            <option value='licensed_to_serve_ages=Pre-Kindergarten'>Pre-Kindergarten</option>
                            <option value='licensed_to_serve_ages=School'>School</option>
                        </Form.Select>
                    </Form.Group>
                </Row>
            </Form>
        </div>
    );
}

export default ChildcareFilterBar;