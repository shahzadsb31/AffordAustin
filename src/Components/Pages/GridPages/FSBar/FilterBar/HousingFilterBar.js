// import './FilterBar.css'
import 'bootstrap/dist/css/bootstrap.min.css'
import React, { useState, useEffect } from 'react';
import { Row, Col, Form } from 'react-bootstrap';

//Change values to queries
const HousingFilterBar = ({sendQuery}) => {
    const [form, setForm] = useState({
        'NumUnitsFilter': '', 
        'TenureFilter': '', 
        'UnitTypeFilter': '', 
        'ZipcodeFilter': '', 
        'GroundLeaseFilter': ''
    });

    const setField = (field, value) => {
        setForm({
            ...form,
            [field]: value
        })
    }

    useEffect(() => {
        let isNotEmpty = form['NumUnitsFilter'] !== '';
        let filterQuery = form['NumUnitsFilter'];

        if ( isNotEmpty && form['TenureFilter'] !== '') {
            filterQuery += '&';
        }
        filterQuery += form['TenureFilter'];
        isNotEmpty = isNotEmpty || form['TenureFilter'] !== '';

        if (isNotEmpty && form['UnitTypeFilter'] !== '') {
            filterQuery += '&';
        }
        filterQuery += form['UnitTypeFilter'];
        isNotEmpty = isNotEmpty || form['UnitTypeFilter'] !== '';

        if (isNotEmpty && form['ZipcodeFilter'] !== '') {
            filterQuery += '&';
        }
        filterQuery += form['ZipcodeFilter'];
        isNotEmpty = isNotEmpty || form['ZipcodeFilter'] !== '';

        if ( isNotEmpty && form['GroundLeaseFilter'] !== '') {
            filterQuery += '&';
        }
        filterQuery += form['GroundLeaseFilter'];
        
        sendQuery(filterQuery);
    }, [form])

    return (
        <div style={{ textAlign:'center' }}>
            <h3>Filters:</h3>
            <Form>
                <Row className="g-3 justify-content-center" xs='auto'>
                    <Form.Group controlId='NumUnitsFilter' as={Col}>
                        <Form.Label>Number of Units</Form.Label>
                        <Form.Select
                            className='filter_select'
                            onChange={e => setField('NumUnitsFilter', e.target.value)}
                        >
                            <option value=''>Select #Units</option>
                            <option value='total_affordable_units=0-5'>&lt;5</option>
                            <option value='total_affordable_units=5-10'>5-10</option>
                            <option value='total_affordable_units=10-50'>10-50</option>
                            <option value='total_affordable_units=50-100'>50-100</option>
                            <option value='total_affordable_units=100'>100+</option>
                        </Form.Select>
                    </Form.Group>
                    <Form.Group controlId='TenureFilter' as={Col}>
                        <Form.Label>Tenure</Form.Label>
                        <Form.Select
                            className='filter_select'
                            onChange={e => setField('TenureFilter', e.target.value)}
                        >
                            <option value=''>Select Tenure</option>
                            <option value='tenure=Rental'>Rental</option>
                            <option value='tenure=Ownership'>Ownership</option>
                        </Form.Select>
                    </Form.Group>
                    <Form.Group controlId='UnitTypeFilter' as={Col}>
                        <Form.Label>Unit Type</Form.Label>
                        <Form.Select
                            className='filter_select'
                            onChange={e => setField('UnitTypeFilter', e.target.value)}
                        >
                            <option value=''>Select Type</option>
                            <option value='unit_type=Single Family'>Single Family</option>
                            <option value='unit_type=Multifamily'>Multifamily</option>
                            <option value='unit_type=Duplex'>Duplex</option>
                            <option value='unit_type=FourPlex'>Fourplex</option>
                            <option value='unit_type=ADU'>ADU</option>
                        </Form.Select>
                    </Form.Group>
                    <Form.Group controlId='ZipcodeFilter' as={Col}>
                        <Form.Label>Zip Code</Form.Label>
                        <Form.Control 
                            type='text'
                            placeholder='Zip Code'
                            className='filter_text'
                            // From https://stackoverflow.com/questions/34223558/enter-key-event-handler-on-react-bootstrap-input-component
                            onKeyPress={e => {
                                if (e.key === "Enter") {
                                    setField('ZipcodeFilter', (e.target.value == "") ? "" : "zip_code=" + e.target.value);
                                }
                            }}
                        />
                    </Form.Group>
                    <Form.Group controlId='GroundLeaseFilter' as={Col}>
                        <Form.Label>Ground Lease</Form.Label>
                        <Form.Select 
                            className='filter_select'
                            onChange={e => setField('GroundLeaseFilter', e.target.value)}
                        >
                            <option value=''>Ground Lease</option>
                            <option value='ground_lease=Yes'>Available</option>
                            <option value='ground_lease=No'>Not Available</option>
                        </Form.Select>
                    </Form.Group>
                </Row>
            </Form>
        </div>
    );
}

export default HousingFilterBar;