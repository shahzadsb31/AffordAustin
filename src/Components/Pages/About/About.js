import React from "react";
import { Row, Container, Button, Card, Col, ListGroup, Image } from "react-bootstrap";
import "bootstrap/dist/css/bootstrap.min.css";
import MemberCards from "./MemberCards/MemberCards";
import GitTotals from "./GitTotals/GitTotals";
import "./About.css";
import * as Icon from "react-bootstrap-icons";
import axios from "axios";
import { useEffect, useState } from "react";
import teamInfo from "./TeamData";
import AWSAmplifyLogo from "./ToolLogos/AWSAmplifyLogo.png"
import AWSLogo from "./ToolLogos/AWSLogo.png"
import BlackLogo from "./ToolLogos/BlackLogo.png"
import DockerLogo from "./ToolLogos/DockerLogo.png"
import FlaskLogo from "./ToolLogos/FlaskLogo.jpg"
import JestLogo from "./ToolLogos/JestLogo.png"
import MarshmallowLogo from "./ToolLogos/MarshmallowLogo.png"
import PostgresLogo from "./ToolLogos/PostgresLogo.png"
import SeleniumLogo from "./ToolLogos/SeleniumLogo.png"
import SQLAlchemyLogo from "./ToolLogos/SQLAlchemyLogo.png"
import GitlabLogo from "./ToolLogos/GitlabLogo.png"
import NameCheapLogo from "./ToolLogos/NameCheapLogo.jpg"
import PostmanLogo from "./ToolLogos/PostmanLogo.png"
import ReactLogo from "./ToolLogos/ReactLogo.png"

const tools = ["React", "Amplify", "GitLab", "NameCheap", "Postman", "AWS", "Black",
               "Docker", "Flask", "Jest", "Marshmallow", "Postgres", "Selenium", 
               "SQLAlchemy",];
const tool_link = ["https://reactjs.org/", "https://aws.amazon.com/amplify/", "https://about.gitlab.com/", "https://www.namecheap.com/", "https://www.postman.com/", 
    "https://aws.amazon.com/about-aws/", "https://github.com/psf/black", "https://docs.docker.com/get-started/overview/",
    "https://flask.palletsprojects.com/en/2.1.x/#", "https://jestjs.io/", "https://github.com/marshmallow-code/marshmallow",
    "https://www.postgresql.org/about/", "https://www.selenium.dev/about/", "https://www.sqlalchemy.org/"];
const tool_logo = [ReactLogo, AWSAmplifyLogo, GitlabLogo, NameCheapLogo, PostmanLogo, AWSLogo,
                   BlackLogo, DockerLogo, FlaskLogo, JestLogo, MarshmallowLogo, PostgresLogo,
                   SeleniumLogo, SQLAlchemyLogo];

const tools_desc = [
  "Used to develop the frontend using UI components.",
  "Managed Amazon Web Services backend used for deploying frontend code.",
  "Contains production site code. Used for seamless and collaborative integration.",
  "Used to acquire a domain name for a useable URL and SSL authentication.",
  "Helped design the backend RESTful API design and documentation.",
  "Used for database and backend API server management.",
  "Used for formatting python code.",
  "Containerization tool used to manage environment images.",
  "Web app framework used for API development.",
  "Used for testing Javascript code.",
  "Used for managing database schemas.",
  "Management system for the relational database.",
  "Used for testing scripts in browser.",
  "Python SQL toolkit and Object relation manager."
];

const apis = [
  [
    "SerpAPI Jobs API",
    "Used to find job postings.",
    "https://serpapi.com/google-jobs-api",
  ],
  [
    "data.texas.gov API",
    "Contains two disparate data sources used for finding metadata related to housing and childcare in Austin.",
    "https://data.texas.gov/",
  ],
  [
    "Google Maps",
    "Provides an iframe to embed Google Maps data.",
    "https://www.google.com/maps",
  ],
  [
    "Google Images",
    "Used to retrieve related image data for model instances.",
    "https://www.google.com/imghp?hl=EN",
  ],
];

let total_issues = 0;
let total_commits = 0;

const contributor_to_team_num = {
  sca: 0,
  din: 1,
  jay: 2,
  pre: 3,
  phe: 3,
  sab: 4,
  sha: 4,
};

function About() {
  const [total_commits, setCommits] = useState(0);
  const [total_issues, setIssues] = useState(0);
  useEffect(() => {
    async function getGitlabStats() {
      await axios
        .get(
          "https://gitlab.com/api/v4/projects/33875511/repository/contributors"
        )
        .then((resp) => {
          let c = 0;
          resp.data.forEach((contributor) => {
            const { name, email, commits } = contributor;
            let team_num =
              contributor_to_team_num[name.substring(0, 3).toLowerCase()];
            teamInfo[team_num]["commits"] = commits;
            c += commits;
          });
          setCommits(c);
        });
      let i = 0;
      Promise.all(
        teamInfo.map(async (member) => {
          let issues_api =
            "https://gitlab.com/api/v4/projects/33875511/issues_statistics?author_username=" +
            member.user;
          await axios.get(issues_api).then((resp) => {
            member["issues"] = resp["data"]["statistics"]["counts"]["closed"];
            i += resp["data"]["statistics"]["counts"]["closed"];
          });
        })
      ).then(() => {
        setIssues(i);
      });
    }
    getGitlabStats();
  }, []);
  return (
    <div
      className="App"
      style={{ backgroundColor: "#f0f2f5", paddingBottom: "3em" }}
    >
      {/* About us block */}
      <h1 className="section_header">About Us</h1>
      <Container>
        <Card
          border="light"
          className="about_text mx-auto"
          style={{ borderRadius: "2rem" }}
        >
          <Card.Body className="" style={{ color: "black" }}>
            The cost of living in Austin has reportedly risen 17.8% since 2010.
            In the past year alone, the listing prices for houses spiked 28%. At
            AffordAustin, we strive to provide plausible living situations in
            Austin amidst harsh inflation by streamlining the process of finding
            economic housing, child care, and work. Users will be able to search
            by any of the aforementioned categories based on their priority,
            then find nearby instances of the other two categories.
          </Card.Body>
        </Card>
      </Container>

      {/* Member Cards */}
      <MemberCards teamInfo={teamInfo} />

      {/*Git Total Stats*/}
      <h1 className="section_header">Git Statistics</h1>
      <GitTotals total_commits={total_commits} total_issues={total_issues} />

      {/*Tool Info*/}
      <h1 className="section_header">Tools Used</h1>
      <Card
       
        className="instance_data mx-auto border-0"
        style={{
          borderRadius: "2rem",
          borderTopLeftRadius: "2rem",
          backgroundColor: "#f0f2f5"
        }}
      >

          <Row xs='auto'className="justify-content-center">
            
            {tools.map((tool, index) => (
              <Card border="light" className="tool_about_list m-4">
                <a className="tool_desc" href={tool_link[index]}>
                <Card.Img variant="top" src={tool_logo[index]} className="tool_logo mt-1" />
                <Card.Body>
                  <Card.Title className="mt-auto" ></Card.Title>
                    <b>{tool}</b>
                  <Card.Text style={{fontSize:"95%"}}>{tools_desc[index]}</Card.Text>
                </Card.Body>
                </a>

              </Card>



                /* <Col className="tool_about_list mx-auto">
                <Row href={tool_link[index]} className="justify-content-center"><Image src={tool_logo[index]} className="tool_logo"></Image></Row>
                <a href={tool_link[index]}><b>{tool}</b></a>
                <br></br>
                {tools_desc[index]}
                <br></br>
                
                </Col> */
            ))}
          </Row>
      </Card>

      
      

      {/*API Info*/}
      <h1 className="section_header">Data Source Links</h1>
      <Card
        border="light"
        className="instance_data mx-auto"
        style={{
          backgroundColor: "#f0f2f5",
          borderRadius: "2rem",
          borderTopLeftRadius: "2rem",
        }}
      >
        <ListGroup
          variant="flush"
          style={{
            borderRadius: "2rem",
          }}
        >
          {apis.map((api) => (
            <a href={api[2]} target="_blank">
              <ListGroup.Item className="link about_list" key={api[0]}>
                <b>{api[0]}</b>
                <br></br>
                {api[1]}
              </ListGroup.Item>
            </a>
          ))}
        </ListGroup>
      </Card>

      <h1 className="section_header">Our Links</h1>
      <Card
        className="instance_data mx-auto border-0"
        style={{
          borderRadius: "2rem",
          borderTopLeftRadius: "2rem",
          backgroundColor: "#f0f2f5"
        }}
      >

          <Row xs='auto'className="justify-content-center">
            
              <Card border="light" className="tool_about_list m-4">
                <a className="tool_desc" href="https://gitlab.com/dinesh.k.balakrishnan/cs373-website">
                <Card.Img src={tool_logo[2]} />
                <Card.Body>
                  <Card.Title className="" ></Card.Title>
                    <b>Gitlab</b>
                  <Card.Text style={{fontSize:"95%"}}> Our Repository</Card.Text>
                </Card.Body>
                </a>
              </Card>

              <Card border="light" className="tool_about_list m-4">
                <a className="tool_desc" href="https://documenter.getpostman.com/view/19702236/UVksLu2r">
                <Card.Img className = "mt-4" style ={{width:"75%", height:"54%"}} src={tool_logo[4]} />
                <Card.Body>
                  <Card.Title className="mt-3" ></Card.Title>
                    <b>Postman</b>
                  <Card.Text style={{fontSize:"95%"}}> Our API Documentation</Card.Text>
                </Card.Body>
                </a>

              </Card>
          </Row>
      </Card>

      <h1 className="section_header">Interesting Results</h1>
      <ul>
        <li>
          <h2 className="temp_interesting">
            The most affordable housing instances are located in Travis County.
          </h2>
        </li>
        <li>
          <h2 className="temp_interesting">
            The most daycare instances are located in Harris County.
          </h2>
        </li>
      </ul>
    </div>
  );
}

export default About;
