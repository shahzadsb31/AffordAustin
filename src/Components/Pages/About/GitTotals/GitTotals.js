import { Row, Container, Card, Col } from "react-bootstrap";
import "bootstrap/dist/css/bootstrap.min.css";
import "./GitTotals.css";
import * as Icon from "react-bootstrap-icons";
import React from "react";

function GitTotals(props) {
  return (
    <>
      <Container>
        {/*style={{ width: "100%" }} add this back in to row if things mess up*/}
        <Row className="justify-content-center g-4" xs="auto">
          <Col xs="auto">
            <Card border="light" style={{ borderRadius: "2rem" }}>
              <Card.Body className="git_totals">
                <Card.Title className="text-truncate mt-1 card_names">
                  Total Commits
                </Card.Title>
                <Icon.Check2Circle
                  className="mt-2"
                  style={{ fontSize: "50px", color: "black" }}
                />
                <Card.Subtitle className="m-2 card_names">
                  {props.total_commits}
                </Card.Subtitle>
              </Card.Body>
            </Card>
          </Col>

          <Col xs="auto">
            <Card border="light" style={{ borderRadius: "2rem" }}>
              <Card.Body className="git_totals">
                <Card.Title className="mt-1 card_names">
                  Total Issues
                </Card.Title>
                <Icon.ListCheck
                  className="mt-2"
                  style={{ fontSize: "50px", color: "black" }}
                />
                <Card.Subtitle className="m-2 card_names">
                  {props.total_issues}
                </Card.Subtitle>
              </Card.Body>
            </Card>
          </Col>

          <Col xs="auto">
            <Card border="light" style={{ borderRadius: "2rem" }}>
              <Card.Body className="git_totals">
                <Card.Title className="text-truncate mt-1 card_names">
                  Total Tests
                </Card.Title>
                <Icon.Wrench
                  className="mt-2"
                  style={{ fontSize: "50px", color: "black" }}
                />
                <Card.Subtitle className="m-2 card_names">89</Card.Subtitle>
              </Card.Body>
            </Card>
          </Col>
        </Row>
      </Container>
    </>
  );
}

export default GitTotals;
