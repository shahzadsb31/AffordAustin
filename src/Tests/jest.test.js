import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

import About from "../Components/Pages/About/About";
import GitTotals from "../Components/Pages/About/GitTotals/GitTotals";
import Home from "../Components/Pages/Home/Home";
import Housing from "../Components/Pages/InstancePages/Housing";
import ChildCare from "../Components/Pages/InstancePages/ChildCare";
import Jobs from "../Components/Pages/InstancePages/Jobs";

import Navbar from "../Components/MainNavBar/MainNavBar";

import ChildCareGrid from "../Components/Pages/GridPages/ChildCareGrid";
import HousingGrid from "../Components/Pages/GridPages/HousingGrid";
import JobGrid from "../Components/Pages/GridPages/JobGrid";

import SearchPage from "../Components/Pages/Search/SearchPage";

import ChildcareFilterBar from "../Components/Pages/GridPages/FSBar/FilterBar/ChildcareFilterBar";
import HousingFilterBar from "../Components/Pages/GridPages/FSBar/FilterBar/HousingFilterBar";
import JobFilterBar from "../Components/Pages/GridPages/FSBar/FilterBar/JobFilterBar";

import ChildcareSortBar from "../Components/Pages/GridPages/FSBar/SortBar/ChildcareSortBar";
import HousingSortBar from "../Components/Pages/GridPages/FSBar/SortBar/HousingSortBar";
import JobSortBar from "../Components/Pages/GridPages/FSBar/SortBar/JobSortBar";

import FSBar from "../Components/Pages/GridPages/FSBar/FSBar";
import Pagination from "../Components/Pages/GridPages/FSBar/Pagination";

configure({ adapter: new Adapter() });

describe("Render components", () => {
  it("1. render About page", () => {
    const tree = shallow(<About />);
    expect(tree).toMatchSnapshot();
  });

  it("2. Home match snapshot", () => {
    const home = shallow(<Home />);
    expect(home).toBeDefined();
  });

  it("3. GitTotals match snapshot", () => {
    const gittotals = shallow(<GitTotals />);
    expect(gittotals).toBeDefined();
  });

  it("4. Housing matches snapshot", () => {
    const tree = shallow(<Housing />);
    expect(tree).toMatchSnapshot();
  });

  it("5. ChildCare matches snapshot", () => {
    const tree = shallow(<ChildCare />);
    expect(tree).toMatchSnapshot();
  });

  it("6. Jobs matches snapshot", () => {
    const tree = shallow(<Jobs />);
    expect(tree).toMatchSnapshot();
  });

  it("7. Navbar matches snapshot", () => {
    const navbar = shallow(<Navbar />);
    expect(navbar).toMatchSnapshot();
  });

  it("11. Searching page matches snapshot", () => {
    const searchpage = shallow(<SearchPage />);
    expect(searchpage).toMatchSnapshot();
  });

  it("12. ChildcareFilterBar matches snapshot", () => {
    const ccfilterbar = shallow(<ChildcareFilterBar />);
    expect(ccfilterbar).toMatchSnapshot();
  });

  it("13. HousingFilterBar matches snapshot", () => {
    const hbar = shallow(<HousingFilterBar />);
    expect(hbar).toMatchSnapshot();
  });

  it("14. JobFilterBar matches snapshot", () => {
    const jbar = shallow(<JobFilterBar />);
    expect(jbar).toMatchSnapshot();
  });

  it("15. ChildcareSortBar matches snapshot", () => {
    const ccfilterbar = shallow(<ChildcareSortBar />);
    expect(ccfilterbar).toMatchSnapshot();
  });

  it("16. HousingSortBar matches snapshot", () => {
    const hbar = shallow(<HousingSortBar />);
    expect(hbar).toMatchSnapshot();
  });

  it("17. JobSortBar matches snapshot", () => {
    const jbar = shallow(<JobSortBar />);
    expect(jbar).toMatchSnapshot();
  });

  it("18. FSBar matches snapshot", () => {
    const fsbar = shallow(<FSBar />);
    expect(fsbar).toMatchSnapshot();
  });

  it("19. Pagination matches snapshot", () => {
    const pag = shallow(<Pagination />);
    expect(pag).toMatchSnapshot();
  });
});
