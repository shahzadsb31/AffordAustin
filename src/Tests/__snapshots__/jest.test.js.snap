// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render components 1. render About page 1`] = `ShallowWrapper {}`;

exports[`Render components 4. Housing matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Render components 5. ChildCare matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Render components 6. Jobs matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Render components 7. Navbar matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Render components 11. Searching page matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Render components 12. ChildcareFilterBar matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Render components 13. HousingFilterBar matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Render components 14. JobFilterBar matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Render components 15. ChildcareSortBar matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Render components 16. HousingSortBar matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Render components 17. JobSortBar matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Render components 18. FSBar matches snapshot 1`] = `ShallowWrapper {}`;

exports[`Render components 19. Pagination matches snapshot 1`] = `ShallowWrapper {}`;
